cupy/core/_kernel.pyx

```diff
@@ -1115,16 +1115,18 @@ cdef class _Ops:
     cdef _Ops from_tuples(object ops, routine):
         ops_ = []
         for t in ops:
-            if isinstance(t, _Op):
-                ops_.append(t)
-            elif isinstance(t, tuple):
+            if isinstance(t, tuple):
                 typ, rt = t
                 if isinstance(rt, tuple):
                     rt = tuple([r1 or r2 for r1, r2 in zip(rt, routine)])
-                ops_.append(_Op.from_type_and_routine(typ, rt))
+                elif not isinstance(rt, str):
+                    # TODO(kataoka): assert isinstance(rt, callable)
```

Comment:
Why is this TODO? (It should be `assert callable(rt)` actually)

Reply:
It seems `assert callable(rt)` cannot be done in `cdef`.

Reply:
Do you mean `assert` or `callable`, or their combination?
`assert` is used in the code already and I personally confirmed `callable` is usable in Cython.

Reply:
`assert callable(obj)` seems to work, too.

Reply:
Sorry.  `assert callable(obj)` works.